packages/core-services/src/LocalBroker.ts

```diff
@@ -111,8 +142,81 @@ export class LocalBroker implements IBroker {
 		return instances.map(({ _id }) => ({ id: _id, available: true }));
 	}
 
+	/**
+	 * Registers services to be started. We're assuming that each service will only have one level of dependencies.
+	 */
+	private registerPendingServices(services: string[] = []): void {
+		for (const service of services) {
+			this.pendingServices.add(service);
+		}
+	}
+
+	/**
+	 * Removes a service from the pending services set.
+	 */
+	private removePendingService(service: string): void {
+		this.pendingServices.delete(service);
+	}
+
+	private async startService(service: ExtendedServiceClass): Promise<void> {
+		const serviceName = service.instance.getName();
+
+		if (typeof service === 'string') {
+			logger.debug(`Service ${serviceName} is not in the services map. Bringing it back to queue`);
+			return;
+		}
+
+		if (service?.isStarted) {
+			logger.debug(`Service ${serviceName} already started`);
+			return;
+		}
+
+		const pendingDependencies = service.dependencies.filter((e) => !this.services.has(e) || !this.services.get(e)?.isStarted);
+		if (pendingDependencies.length > 0) {
+			logger.debug(
+				`Service ${serviceName} has dependencies that are not started yet, bringing it back to queue: ${pendingDependencies.join(', ')}`,
+			);
+			return;
+		}
+
+		await service.instance.started();
+		this.services.set(serviceName, { ...service, isStarted: true });
+		this.removePendingService(serviceName);
+
+		logger.debug(`Service ${serviceName} successfully started`);
+	}
+
 	async start(): Promise<void> {
-		await Promise.all([...this.services].map((service) => service.started()));
-		this.started = true;
+		const startTime = Date.now();
+
+		return new Promise((resolve, reject) => {
+			const intervalId = setInterval(async () => {
+				const elapsed = Date.now() - startTime;
+
+				const availableServices = Array.from(this.services.values()).filter((service) => service.isStarted);
+
+				if (this.pendingServices.size === 0) {
+					logger.debug(`All services available: ${availableServices.map((s) => s.instance.getName()).join(', ')}`);
+					clearInterval(intervalId);
+					return resolve();
+				}
```

Comment:
there is a log already with every service name, so this looks not so relevant at this point and too noisy as well.

I think it could also be a `info` log here ;)

```suggestion
				if (this.pendingServices.size === 0) {
					const availableServices = Array.from(this.services.values()).filter((service) => service.isStarted);

					logger.info(`All ${availableServices.length} services available`);
					clearInterval(intervalId);
					return resolve();
				}
```